### PR TITLE
Upgrades leak canary, testing, androidx lifecycle.

### PR DESCRIPTION
**Pull request by rjrjr**
Note that we hold off on upgrading androidx activity, appcompat, etc. Those push androidx core past 1.6.0, which breaks https://github.com/cashapp/paparazzi. We should hold off until they catch up, which should happen soon after AS Bumblebee freezes.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -39,6 +39,8 @@ subprojects {
       // There could be transitive dependencies in tests with a lower version. This could cause
       // problems with a newer Kotlin version that we use.
       resolutionStrategy.force(Dependencies.Kotlin.reflect)
+      resolutionStrategy.force(Dependencies.AndroidX.Lifecycle.viewModel)
```
Comment (rjrjr) [inline build.gradle.kts:42]: Nope.

**Comment by rjrjr** [comment]
> But not test dependencies, b/c LeakCanary's hacky test runner breaks, and the [non-hacky alternative](https://github.com/square/leakcanary/pull/2244/) hasn't been released yet.

@pyricau accurate?

**Comment by rjrjr** [comment]
> > But not test dependencies, b/c LeakCanary's hacky test runner breaks, and the [non-hacky alternative](https://github.com/square/leakcanary/pull/2244/) hasn't been released yet.
> 
> @pyricau accurate?

Nope.

File: .buildscript/android-ui-tests.gradle [hunk 2]
```diff
@@ -1,7 +1,6 @@
 android {
   defaultConfig {
     testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
-    testInstrumentationRunnerArgument "listener", "leakcanary.FailTestOnLeakRunListener"
```
Comment (vgonda) [inline .buildscript/android-ui-tests.gradle:4]: I'm assuming removing this was somehow related to the upgrades?

Comment (vgonda) [reply .buildscript/android-ui-tests.gradle:4]: Ahhh it's because we're now chaining `DetectLeaksAfterTestSuccess` with the test rule.

**Review by vgonda — APPROVED** [review]
I'm on board with upgrades

**Review by RBusarow — APPROVED** [review]

**Comment by rjrjr** [comment]
This is consistent failing:

```
* What went wrong:
Execution failed for task ':trace-encoder:kaptGenerateStubsKotlin'.
> Error while evaluating property 'filteredArgumentsMap' of task ':trace-encoder:kaptGenerateStubsKotlin'
   > Could not resolve all files for configuration ':trace-encoder:kapt'.
      > Could not download guava-30.1.1-jre.jar (com.google.guava:guava:30.1.1-jre)
         > Could not get resource 'https://repo.maven.apache.org/maven2/com/google/guava/guava/30.1.1-jre/guava-30.1.1-jre.jar'.
            > Could not GET 'https://repo.maven.apache.org/maven2/com/google/guava/guava/30.1.1-jre/guava-30.1.1-jre.jar'.
               > Connection reset
```

Needless to say we have no direct dependencies on guava, so I imagine we have a friend at Google who is pulling it for us transitively. Good times.

**Comment by rjrjr** [comment]
> This is consistently failing:
> 
> ```
> * What went wrong:
> Execution failed for task ':trace-encoder:kaptGenerateStubsKotlin'.
> > Error while evaluating property 'filteredArgumentsMap' of task ':trace-encoder:kaptGenerateStubsKotlin'
>    > Could not resolve all files for configuration ':trace-encoder:kapt'.
>       > Could not download guava-30.1.1-jre.jar (com.google.guava:guava:30.1.1-jre)
>          > Could not get resource 'https://repo.maven.apache.org/maven2/com/google/guava/guava/30.1.1-jre/guava-30.1.1-jre.jar'.
>             > Could not GET 'https://repo.maven.apache.org/maven2/com/google/guava/guava/30.1.1-jre/guava-30.1.1-jre.jar'.
>                > Connection reset
> ```
> 
> Needless to say we have no direct dependencies on guava, so I imagine we have a friend at Google who is pulling it for us transitively. Good times.

Bah, this is happening a lot on master too, right after green PR runs. Something is rotten in GHA land, as usual.

**Comment by rjrjr** [comment]
Going to hold off on a lot of this. Bumping appcompat also bumps core past 1.6.0, which breaks https://github.com/cashapp/paparazzi. Shouldn't do that w/o a compelling need.

@jrodbx expects to be able to address that once AS Bumblebee is released.

**Pull request merged**